server/the-usher-openapi-spec.yaml

```diff
@@ -467,28 +467,89 @@ paths:
                 $ref: "#/components/schemas/Persona"
         400:
           $ref: '#/components/responses/BadRequest'
-        409: 
+        409:
           $ref: '#/components/responses/Conflict'
         500:
           $ref: '#/components/responses/InternalError'
 
+  /clients:
+    post:
+      'x-swagger-router-controller': 'clients/index'
+      operationId: createClient
+      summary: Create a new Client
+      tags:
+        - Client Admin APIs
+      security:
+        - bearerAdminAuth: []
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                tenant_name:
+                  type: string
+                client_id:
+                  type: string
+                  maxLength: 50
+                  pattern: ^[a-z0-9-_]*$
+                name:
+                  type: string
+                  maxLength: 50
+                description:
+                  type: string
+                secret:
+                  type: string
+              required:
+                - tenant_name
+                - client_id
+                - name
+      responses:
+        201:
+          description: The newly created Client object
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Client'
+        400:
+          $ref: '#/components/responses/BadRequest'
 
   /clients/{client_id}:
+    parameters:
+      - $ref: '#/components/parameters/clientIdPathParam'
```

Comment:
Although we have 
```sql
CONSTRAINT clients_client_id_uq UNIQUE (client_id)
```
But isn't is better to rely on `clientKey` for the sake of consistency across the APIs?

Reply:
@k1ch  this is the API that I was talking about in our conversation yesterday when discussing the use of client id vs. key. 

These APIs follow the existing convention that was from the start of the project of using client id as the "identifier". 
This is what I was asking if we should review and merge this PR (assuming everything else is ok barring the id vs. key) then as a separate change consider the question of client id vs. key.

Reply:
I am on board with the usage of `client_id` as the unique identifier. Additionally, I suggest we drop the `key` column, make `client_id` the primary key, and update the corresponding foreign keys. Having two unique identifiers in a table does not make sense to me

Reply:
@k1ch I see where you are going with the thought, but in this case I am not sure the refactor is worth the effort. 

Logically, having a "database unique key" and a different "Human / User unique key" is duplicative I don't think its the end of the world. If we were designing the schema from scratch we might do it differently.